crates/net/downloaders/src/file_client.rs

```diff
@@ -115,7 +116,10 @@ impl FileClient {
     /// Clones and returns the highest header of this client has or `None` if empty. Seals header
     /// before returning.
     pub fn tip_header(&self) -> Option<SealedHeader> {
-        self.headers.get(&self.max_block()?).map(|h| h.clone().seal_slow())
+        self.headers.get(&self.max_block()?).map(|h| {
+            let sealed = h.clone().seal_slow();
+            SealedHeader::new(sealed.inner().clone(), sealed.seal())
+        })
```

Comment:
why does this have to be changed?

Reply:
Because in alloy `seal_slow` returns a generic `Sealed<T>` type which is different from the `SealedHeader` that we have in reth. So that to be compatible with reth at the moment we have to reconstruct a reth `SealedHeader` from the alloy `Sealed<Header>` type.

But I'm thinking about a follow up PR to replace our reth `SealedHeader` by the alloy `Sealed<Header>` primitive wdyt?

Reply:
this now has an additional clone, this is not great and just weird.
we don't need the seal function here if we only need the hash

Reply:
this can also be solved with a `SealedHeader:: seal(Header)` function

Reply:
Maybe to reduce overhead, what would you say to replace also in this PR `SealedHeader` by alloy `Sealed` type?

Reply:
```suggestion
        self.headers.get(&self.max_block()?).map(|h| {
            let sealed = h.clone().seal_slow();
            let (header, seal) = sealed.into_parts();
            SealedHeader::new(header, seal);
        })
```

I think we can just do this?

Reply:
Just fixed this but I think that in a follow up we can simply rm `SealedHeader` from reth and use `Sealed` instead, it will simplify everything no?